docs/guides/05-legends-and-widgets.md

```diff
@@ -0,0 +1,240 @@
+    <iframe
+        id="legend-buckets"
+        src="/developers/carto-vl/examples/maps/misc/legends/legend-buckets.html"
+        width="100%"
+        height="500"
+        style="margin: 20px auto !important"
+        frameBorder="0">
+    </iframe>
+</div>
+<a href="/developers/carto-vl/examples/#example-legends---categorical-data">View my source code!</a>
+
+##### Showing images
+
+<div class="example-map">
+    <iframe
+        id="legend-image"
+        src="/developers/carto-vl/examples/maps/misc/legends/legend-image.html"
+        width="100%"
+        height="500"
+        style="margin: 20px auto !important"
+        frameBorder="0">
+    </iframe>
+</div>
+<a href="/developers/carto-vl/examples/#example-multiple-images">View my source code!</a>
+
+### Widgets
+
+The first thing you need to know to create widgets is the concept of [variables](). A variable is a name in a CARTO VL visualization that has an expression bound to it. It's simply a way to name expressions to be referenced later. For example, let's say there was a magic function that returned our widget information like `myAwesomeWidgetExpression()`, how do we actually get that information?
+
+Firstly, we'll need to bind that expression to a new name so our visualization could be created like this:
+
+```js
+const myViz = new carto.Viz(`
+    color: red
+    width: 7
+
+    @myAwesomeVariable: myAwesomeWidgetExpression()
+`);
+```
+
+Then, we'll need to ask for the value of the expression when needed. Since our awesome widget expression may change its value dynamically, for example, due to panning or filtering we should probably read it each time its layer is changed:
+
+```js
+myLayer.on('updated', ()=>{
+    const myWidgetData = myViz.variables.myAwesomeVariable.value;
+    updateMyAwesomeWidget(myWidgetData);
+})
+```
+
+If, however, myAwesomeWidgetExpression was static, we could just use `myLayer.on('loaded', ()=>{...});`.
+
+#### Scalars: what is the total of ...? what is the average of ...? what is the maximum ...?
+
+##### What is the average price in the entire dataset?
+
+To get the average of a property in the entire dataset we'll need to use the [`globalAvg()`](https://carto.com/developers/carto-vl/reference/#cartoexpressionsglobalavg) expression.
+
+There are similar expressions for getting the minimum, the maximum, the sum, and percentiles.
+
+##### What is the average price in the features shown?
+
+There are similar functions that only take the features on the screen into account. For example, see [viewportAvg](https://carto.com/developers/carto-vl/reference/#cartoexpressionsviewportavg).
+
+All the _viewport*_ like expressions will take only the features that are on the viewport region and that pass the filter set in the `filter:` visualization property.
+
+You can see all together on this example:
+
+<div class="example-map">
+    <iframe
+        id="accidents-widgets"
+        src="/developers/carto-vl/examples/maps/guides/ramp/accidents-widgets.html"
```

Comment:
Some styling issue in the copy

Reply:
What do you mean?

Reply:
Ah, that thing... Those are my design skills... :sweat_smile:  I can ask for help to Design unless you know how to quick fix it